pkg/api/entries.go

```diff
@@ -303,7 +303,24 @@ func GetLogEntryByUUIDHandler(params entries.GetLogEntryByUUIDParams) middleware
 	switch resp.status {
 	case codes.OK:
 	case codes.NotFound:
-		return handleRekorAPIError(params, http.StatusNotFound, fmt.Errorf("grpc error: %w", resp.err), "")
+		// TODO: remove later
```

Comment:
Instead of doing this logic here, I think it would make more sense in this conditional where we don't have a tree ID:

https://github.com/sigstore/rekor/blob/main/pkg/api/entries.go#L283-L290

so basically, if we don't have a tree id:
1. Check the active tree ID
2. If that fails, start going through each inactive shard & check those.
3. If none of those work, return an error